docs/docs/data-storage-redux.md

```diff
@@ -1,9 +1,55 @@
 ---
 title: Data Storage (Redux)
-issue: https://github.com/gatsbyjs/gatsby/issues/21068
 ---
 
-This is a stub. Help our community expand it.
+During Gatsby's bootstrap & build phases, the state is stored and manipulated using the [Redux](https://redux.js.org/) library. The key purpose of using Redux is to centralize all of the state logic. Reviewing the Gatsby [reducers](https://github.com/gatsbyjs/gatsby/tree/80acb8d5d67f7e277ce44158b36da84d262e5b23/packages/gatsby/src/redux/reducers) and [actions](https://github.com/gatsbyjs/gatsby/tree/80acb8d5d67f7e277ce44158b36da84d262e5b23/packages/gatsby/src/redux/actions) folders gives a comprehensive picture of what state manipulations are possible.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Store
+
+The namespaces in Gatsby's Redux store are a great overview of the Gatsby internals, here are a few:
+
+- [Pages](https://www.gatsbyjs.org/docs/gatsby-internals-terminology/#redux-pages-namespace)
+- [Components](https://www.gatsbyjs.org/docs/gatsby-internals-terminology/#redux-components-namespace)
+- Nodes
+- Schema
+- Jobs
+- Webpack
+- Metadata
+
+The Gatsby [/redux index file](https://github.com/gatsbyjs/gatsby/tree/80acb8d5d67f7e277ce44158b36da84d262e5b23/packages/gatsby/src/redux/index.ts) has two key exports, `store` and `emitter`. Throughout the bootstrap and build phases, `store` is used to get the current state and dispatch actions, while `emitter` is used to register listeners for particular actions. The store is also made available to Gatsby users through the [Node APIs](https://www.gatsbyjs.org/docs/node-apis/).
+
+## Actions
+
+Actions dispatched can have a series of effects as different reducers pick them up. Gatsby augments the typical Redux reducer with a pubsub [mitt](https://www.npmjs.com/package/mitt) `emitter` which subscribes to the Redux store to perform further side-effects.
+
+Actions dispatched in the store cause state changes through the reducers and also trigger listeners registered for that action on the `emitter`. The `subscribe` Redux store method is typically used to connect a web framework like React, Gatsby only uses the `subscribe` method to connect the `emitter`.
+
+The [Gatsby actions](/docs/actions/) are all either internal, public or restricted. The public actions, and a context relevant subset of the restricted actions, are available to users through the [Node APIs](https://www.gatsbyjs.org/docs/node-apis/). 
+
+### Action Journey
+
+At the time of writing there are 10 places in the code where `CREATE_NODE` type actions are dispatched, and 10 places where they are acted on through either reducers or `emitter` listeners.
```

Comment:
I feel like this whole section is going to be outdated fast. Is there a way to write about it more generally that doesn't depend so closely on the source code as it's known now?

Reply:
Yeh, definitely. I've taken out the direct file/line references, made the opening sentence a little less specific and gone a little less deep in the binding step.

It's still dependent on the createRedirect action but I think any way it's written I'd be putting examples in. What do you think?

Reply:
It's improved quite a bit from a maintainability standpoint as long as the createRedirect process doesn't change too much! I added some review suggestions to make it clear that it's an example.